src/explorer/models/classNode.ts

```diff
@@ -12,15 +13,24 @@ export class ClassNode extends NodeBase {
     const displayName: string = this.label;
     const itemUri = DocumentContentProvider.getUri(this.fullName, this.workspaceFolder, this.namespace);
     const isLocalFile = itemUri.scheme === "file";
+    const showServerCopy: boolean = config("explorer.alwaysShowServerCopy", this.workspaceFolder);
+    const serverCopyUri = DocumentContentProvider.getUri(
+      this.fullName,
+      this.workspaceFolder,
+      this.namespace,
+      undefined,
+      undefined,
+      true
+    );
 
     return {
       collapsibleState: vscode.TreeItemCollapsibleState.None,
       command: {
-        arguments: [itemUri],
+        arguments: [isLocalFile && !showServerCopy ? itemUri : serverCopyUri],
         command: "vscode-objectscript.explorer.open",
         title: "Open Class",
       },
-      resourceUri: isLocalFile ? itemUri : undefined,
+      resourceUri: isLocalFile && !showServerCopy ? itemUri : undefined,
       contextValue: "dataNode:classNode",
       label: `${displayName}`,
       tooltip: isLocalFile ? undefined : this.fullName,
```

Comment:
Should this also take account of the new setting?

Reply:
The current behavior is to only attach a resource URI when the file is found locally. If we want to change that I think it should be a separate PR.

Reply:
I was commenting on the line that conditionally sets tooltip

Reply:
Sorry, I misread that. Yes, I'll fix that.